kotest-assertions/kotest-assertions-shared/src/jvmMain/kotlin/io/kotest/data/TableFile.kt

```diff
@@ -0,0 +1,71 @@
+package io.kotest.data
+
+import io.kotest.matchers.shouldBe
+import java.io.File
+
+// TODO I'm only supporting table with 3 arguments until the API stabilizes
+
+fun <A, B, C> table(
+   headers: Headers3,
+   file: File,
+   transform: (String, String, String) -> Row3<A, B, C>
+): Table3<A, B, C> {
+   val rows = file.readStringTable(headers.values())
+      .mapRows { (a, b, c) -> transform(a, b, c) }
+   return Table3(headers, rows)
+}
+
+fun File.writeTable(headers: Headers1, rows: List<Row1<String>>): String =
+   writeTable(headers.values(), rows.map { it.strings() } )
+fun File.writeTable(headers: Headers2, rows: List<Row2<String, String>>): String =
+   writeTable(headers.values(), rows.map { it.strings() } )
+fun File.writeTable(headers: Headers3, rows: List<Row3<String, String, String>>): String =
+   writeTable(headers.values(), rows.map { it.strings() } )
+// TODO
+
+private fun Row.strings(): List<String> = values().map { it.toString() }
+
+
+internal fun File.readStringTable(headers: List<String>): StringTable {
+   if (exists().not()) throw AssertionError("Can't read table file")
+   if (extension != "table") throw AssertionError("Table file must have a .table extension")
+   val lines = readLines()
+   if (lines.isEmpty()) throw AssertionError("Table file must have a header")
+   StringTable.parseRow(lines.first()) shouldBe headers
+   return StringTable(headers, lines, skipFirstLine = true)
+}
+
+fun File.writeTable(headers: List<String>, cells: List<List<String>>): String {
```

Comment:
TODO: it would be nice if it was possible to programmatically add comments and new lines, maybe something like this:

```kotlin
val comments = mapOf(1 to "comment at first line", 5 to "", 6 to "other comment", rows.lastIndex to "That's all I've got")
file.writeTable(headers, rows, comments)
```